Makefile

```diff
@@ -74,26 +74,43 @@ endif
 # export LSAN_OPTIONS=suppressions=.asanignore
 AFLAGS = -fsanitize=address #-fsanitize=undefined -fno-omit-frame-pointer
 
-CC_VENDOR := $(firstword $(filter gcc clang,$(shell $(CC) --version)))
+CC_VENDOR := $(firstword $(filter gcc clang icc XL,$(shell $(CC) --version)))
+FC_VENDOR := $(firstword $(filter GNU ifort XL,$(shell $(FC) --version 2>&1 || $(FC) -qversion)))
+
+# Default extra flags by vendor
+MARCHFLAG.gcc           := -march=native
+MARCHFLAG.clang         := $(MARCHFLAG.gcc)
+MARCHFLAG.icc           :=
+OMP_SIMD_FLAG.gcc       := -fopenmp-simd
+OMP_SIMD_FLAG.clang     := $(OMP_SIMD_FLAG.gcc)
+OMP_SIMD_FLAG.icc       := -qopenmp-simd
+OPT.gcc                 := -ffp-contract=fast
+OPT.clang               := $(OPT.gcc)
+CFLAGS.gcc              := -fPIC -std=c99 -Wall -Wextra -Wno-unused-parameter -MMD -MP
+CFLAGS.clang            := $(CFLAGS.gcc)
+CFLAGS.icc              := $(CFLAGS.gcc)
+CFLAGS.XL               := -qpic -MMD
+CXXFLAGS.gcc            := -fPIC -std=c++11 -Wall -Wextra -Wno-unused-parameter -MMD -MP
+CXXFLAGS.clang          := $(CXXFLAGS.gcc)
+CXXFLAGS.icc            := $(CXXFLAGS.gcc)
+CXXFLAGS.XL             := -qpic -MMD
+FFLAGS.GNU              := -fPIC -cpp -Wall -Wextra -Wno-unused-parameter -Wno-unused-dummy-argument -MMD -MP
+FFLAGS.ifort            := -fPIC -cpp
+FFLAGS.XL               := -qpic -ffree-form -qpreprocess -qextname -MMD
 
 cc_check_flag = $(shell $(CC) -E $(1) -x c /dev/null > /dev/null 2>&1 && echo 1)
-MARCHFLAG := -march=native
+MARCHFLAG := $(MARCHFLAG.$(CC_VENDOR))
 MARCHFLAG := $(if $(call cc_check_flag,$(MARCHFLAG)),$(MARCHFLAG),-mcpu=native)
```

Comment:
Do we always want to check for `-mcpu=native` as a backup?

Reply:
Well, XL crashes if we pass it, so it isn't much of a backup. Note that `MARCHFLAG.XL` is empty, and thus is expected to pass (and xl apparently ignores all other arguments when using `-E`).

`gcc` actually errors on Lassen with `-march=native`, but `clang` only warns, thus I should add `-Werror` to `cc_check_flag`.